resources/docker-compose.yml

```diff
@@ -11,18 +11,18 @@ services:
       - 50000:50000
     container_name: jenkins
     environment:
-      - JENKINS_JAVA_OPTS="-Xmx8g"
+      - JENKINS_JAVA_OPTS="-Xmx18g"
       - CASC_RELOAD_TOKEN=reloadPasswordHere
     volumes:
       - /var/lib/jenkins:/var/jenkins_home
     deploy:
       resources:
         limits:
-          cpus: '4'
-          memory: '10g'
+          cpus: '12'
+          memory: '20g'
         reservations:
-          cpus: '4'
-          memory: '10g'
+          cpus: '12'
+          memory: '20g'
```

Comment:
These values are not considered by the system. I saw a warning that this will be discarded
```
The following deploy sub-keys are not supported and have been ignored: resources.reservations.cpus
Creating network "default_default" with the default driver
Pulling jenkins (opensearchstaging/jenkins:2.332.3-lts-jdk8)...
2.332.3-lts-jdk8: Pulling from opensearchstaging/jenkins
Digest: sha256:19371fc44e5988d2b80c0e5847c31537f27d9e1da4c576ace6c02f7cac0a631f
Status: Downloaded newer image for opensearchstaging/jenkins:2.332.3-lts-jdk8
Creating jenkins ...
Creating jenkins ... done
```

Reply:
Version:
```
docker-compose version
docker-compose version 1.29.2, build unknown
docker-py version: 5.0.3
CPython version: 3.7.10
OpenSSL version: OpenSSL 1.0.2k-fips  26 Jan 2017

Reply:
Its the version of `docker-compose`, I see one more problem https://pypi.org/project/docker-compose/
[pip3 install docker-compose](https://github.com/opensearch-project/opensearch-ci/blob/main/lib/compute/jenkins-main-node.ts#L239), gives an old one, we should change that as well.

Reply:
Stack is up with 
```
docker-compose --version
Docker Compose version v2.9.0
```